### PR TITLE
Fix tutorials on Windows

### LINKED ISSUE #549: Windows support in OpenFL tutorials
Hello,

I've posted this below as discussion but since I've just updated my post, I figured it would fit as an issue now related to supporting Windows.

### Discussed in https://github.com/intel/openfl/discussions/546

<div type='discussions-op-text'>

<sup>Originally posted by **Maxime-Perret** October 26, 2022</sup>
Hello,

I've been experimenting with OpenFL trying to run different tutorials. Since supporting Windows was mentioned in #525, I am personally using Windows 11 Pro and ran into two problems which I believe are OS related in the different tutorials.

- For `DataLoader`, having `num_workers` any larger than `0` causes an issue [as such](https://stackoverflow.com/questions/71713719/runtimeerror-dataloader-worker-pids-15876-2756-exited-unexpectedly). Thus, for Windows users, having `num_workers = 0` would be necessary. As far as I know, out of the [interactive api folder](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api), this concerns the tutorials [Dogs vs Cats](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_DogsCats_ViT), [Histology](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Histology), [Histology FedCurv](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Histology_FedCurv), [Kvasir UNet](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Kvasir_UNet), [Lightning MNIST GAN](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Lightning_MNIST_GAN), [MVTec PatchSVDD](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MVTec_PatchSVDD), [Market Re-ID](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_Market_Re-ID), [MedMNIST 2D](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MedMNIST_2D), [MedMNIST 3D](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/PyTorch_MedMNIST_3D).

- Some of the arguments in the plans of the tutorials (in `workspace/plan/plan.yaml`) contain `!!`, like for example `criterion: !!python/object:torch.nn.modules.loss.CrossEntropyLoss`. When parsing the plan in [`federated/plan/plan.py`](https://github.com/intel/openfl/blob/develop/openfl/federated/plan/plan.py), line 42 uses `safe_load` which together causes an issue [`could not determine a constructor for the tag`](https://death.andgravity.com/yaml-unknown-tag). Using the fix suggested [here](https://github.com/yaml/pyyaml/issues/266), I got it to work by replacing `safe_load(yaml_path.read_text())` by `load(yaml_path.read_text(), Loader=yaml.Loader)`.

- Using the `fx pki install` command downloads the binaries for `step` and `step_ca`. The code to download those filters the available assets on github by `content_type` to be `applications/gzip` in `openfl/component/ca/ca.py`, l.74. However, while the Linux assets are `.tar.gz`, the Windows assets are `.zip`, with `content_type` then being `application/zip`. This restriction makes it impossible to find assets for Windows. I tried forcing it to accept but it still fails later on regardless.

Best

Edit: Updated my post after fixing some of my problems with a clean reinstall of Python+Anaconda</div>

### PR DESCRIPTION
**Pull request by itrushkin**
- PyTorch DataLoader's `num_workers` usage is removed due to Windows errors/slowness with `num_workers > 0`.
Closes #549.

### CODE REVIEW

**Comment by itrushkin** [comment]
The merge of this PR may also fix the [MemoryError on Windows](https://github.com/intel/openfl/actions/runs/4075961472/jobs/7023056698#step:5:514).

**Review by igor-davidyuk — APPROVED** [review]
Looks good!

**Review by aleksandr-mokrov — APPROVED** [review]

**Review by mansishr — CHANGES_REQUESTED** [review]
LGTM, but could we add this to the documentation that `num_workers>0` can be used on ubuntu so that users have that option?

**Comment by itrushkin** [comment]
> LGTM, but could we add this to the documentation that `num_workers>0` can be used on ubuntu so that users have that option?

@mansishr I like the idea of documenting the ability to speed up the data loading but the `num_workers` issue is not related to OpenFL. It appears even in non-federated usage of `DataLoader`. Users always have the option of setting the `num_workers` because `torch.utils.data.DataLoader` class accepts it. Why do we need to tell users how to use PyTorch?

**Comment by theakshaypant** [comment]
@itrushkin can you please resolve the comments/conflicts and request for review again?

**Comment by MasterSkepticista** [comment]
Interactive API is deprecated @theakshaypant. I suggest closing this PR, unless there are open issues with Windows builds. Are there?

**Comment by theakshaypant** [comment]
> Interactive API is deprecated @theakshaypant. I suggest closing this PR, unless there are open issues with Windows builds. Are there?

@MasterSkepticista None that I am aware of. The [Windows CI](https://github.com/securefederatedai/openfl/actions/workflows/windows.yml) is all green. 
I agree that we should close the PR.
